### PR TITLE
hasCard, waitingForCard statuses exposed over API + some logging

### LINKED ISSUE #12: insertedCard behavior (WMMT3DX+)
Doing some test runs on actual hardware I think I noticed that if you use game to create new card emulator will still report (`GET /api/v1/insertedCard`) as if no card was inserted. Technically it's true as card was _dispensed_, not _inserted_, but for purposes of creating NFC replacement I'd like to know if card is present in reader or not.

I.e. my idea of NFC replacement is as it follows:

* by default emulator is configured to write into "dummy" card in case someone plays the game and requests a card without using an rfid/nfc card beforehand.*
* tapping nfc reader with valid card tells emulator to "insert" card associated with its serial number
* nfc controller polls emulator to see if card is present inside emulated reader, if it's not present anymore (i.e. it was ejected) it resets the emulator to dummy card

---
*) i checked, changing dispenser status while game is running is a bad idea - if you set dispenser to empty while it was full before game will error out and require reset, if you set it to full while it was empty before it will still act as if it's empty.

### PR DESCRIPTION
**Pull request by hjri**
I cannot into C++, tried passing `cardHandler` and using its `HasCard()` directly but i don't know C++ enough to make it work, therefore I just added `hasCard` variable/property into `CardIO::Settings` that is only updated via moving card. Similarly added `waitingForCard` to show whether card reader is in `WAITING_FOR_CARD` mode, might be a bit janky since i'm not entirely sure when it's supposed to be reset, but seems to be working on WMMT3DX+ at least.

`POST insertedCard` now also has `loadonly` parameter so that you can tell emulator to load specific file without "inserting" it - this way you can prime the emulator to writing into specific file while game creates new card, trying to forcefully "insert" a card while game tries to make new card will result in error.

Also adds some logging to see what's going on with emulator without having to enable debugging/poll the API.

Fixes #12 and #13

### CODE REVIEW

**Comment by GXTX** [comment]
Since the net portion is getting too large I wanted to move it to it's own class/file but i've been quite busy. If this is functional I'll merge now.

**Comment by hjri** [comment]
It's functional, yeah. I'm still not sure how to properly handle the "shutter open" status, and i forgot to test "no card got inserted within time frame" situation back when i was originally testing it.

File: Includes/CardIo.cpp [hunk 1]
```diff
@@ -685,6 +687,10 @@ void CardIo::HandlePacket()
 	UpdateRStatus();
 
 	if (runningCommand) { // 20 is a special case, see function
+		if (cardSettings.waitingForCard) {
```
Comment (GXTX) [inline Includes/CardIo.cpp:690]: This could get spammy in the situation where the test menu will wait infinitely for a card.

**Comment by GXTX** [comment]
Thanks!

**Pull request merged**